src/nvim/message.c

```diff
@@ -2383,6 +2383,15 @@ static void redir_write(char_u *str, int maxlen)
   char_u      *s = str;
   static int cur_col = 0;
 
+  // Append output to capture().
+  if (capture_ga) {
+    int len = maxlen > 0 ? maxlen : (int)STRLEN(str);
```

Comment:
This is `size_t`. Remove cast below in `memmove`, remove it near `STRLEN`, add near `maxlen`.